src/poliastro/maneuver.py

```diff
@@ -306,3 +306,68 @@ def get_total_cost(self):
         """
         dvs = [norm(dv) for dv in self._dvs]
         return sum(dvs, 0 * u.km / u.s)
+
+    @classmethod
+    @u.quantity_input(max_delta_r=u.km)
+    def correct_pericenter(cls, orbit, max_delta_r):
+        """ Return the time before burning and the vector velocity in direction of the burn.
+
+        Parameters
+        ----------
+        orbit : Orbit
+            Position and velocity of a body with respect to an attractor
+            at a given time (epoch).
+        max_delta_r: ~astropy.units.Quantity
+            Maximum satellite’s geocentric distance
+
+        Returns
+        -------
+        delta_t: ~astropy.units.Quantity
+            Maximum time before we must do an orbit-adjustment burn to restore the perigee to its nominal value. (s)
+        vf_ : ~numpy.array
+            Velocity vector of the spacecraft to achieve the desired correction for restoring the perigee to its nominal
+            value. (km/s)
+
+        Raises
+        ------
+        ValueError
+            - If this orbit has no periapsis drift
+            - if the eccentricity is greater than 0.001
+
+        Notes
+        -----
+        The algorithm was obtained from "Fundamentals of Astrodynamics and Applications, 4th ed (2013)" by David A.
+        Vallado, page 885.
+        Given a max_delta_r, we determine the maximum perigee drift before we do an orbit-adjustment burn
+        to restore the perigee to its nominal value. We estimate the time until this burn using the allowable drift
+        delta_w and the drift rate |dw|.
+        For positive delta_v, the change in the eccentricity is positive for perigee burns and negative for apogee burns.
+        The opposite holds for a delta_v applied against the velocity vector, which decreases the satellite’s velocity.
+        Perigee drift are mainly due to the zonal harmonics, which cause variations in the altitude by changing the
+        argument of perigee.
+        Please note that ecc ≈ 0.001, so the error incurred by assuming a small eccentricity is on the order of 0.1%.
+        This is smaller than typical variations in thruster performance between burns.
+
+        """
+        J2 = orbit.attractor.J2.value
+        if J2 == 0.0:
+            raise ValueError("This orbit has no Periapsis drift")
+        elif orbit.ecc > 0.001:
+            raise ValueError(
+                "This orbit has no Periapsis drift, eccentricity should be less than or equal to 0.001"
```

Comment:
What about a `NotImplementedError` instead? There might be actual periapsis drift, but we don't have an analytical expression for the correction maneuver.